File: editorial/chap4-3/prob4-3-3.cpp
#include <iostream>
using namespace std;

int main() {
	double l = 1.0; // √2 を求めたいから
	double r = 2.0; // 初期値を適当に 2.0 にセットする
	
	for (int i = 1; i <= 20; i++) {
		double m = (l + r) / 2.0;
		if (m * m < 2.0) l = m;
		else r = m;
		printf("Step #%d: m = %.12lf¥n", i, m);
	}
	return 0;
}

File: editorial/chap4-3/prob4-3-3.py
l = 1.0
r = 2.0
repeats = 20

for i in range(1, repeats + 1):
	m = (l + r) / 2.0
	if m * m < 2.0:
		l = m
	else:
		r = m
	print("Step #%d: m = %.12f" % (i, m))